service/src/java/org/apache/hive/service/cli/operation/SQLOperation.java

```diff
@@ -258,6 +258,7 @@ public void runInternal() throws HiveSQLException {
         HiveConf.ConfVars.HIVE_SERVER2_ASYNC_EXEC_ASYNC_COMPILE);
     if (!asyncPrepare) {
       prepare(queryState);
+      getParentSession().updateQueryId(this, queryState.getQueryId());
```

Comment:
In fact QueryState.getQueryId is invoked during compile phase before we update it with new one and the old one is stored/used by other objects such as WmContext, QueryDisplay and Operation.createOperationLog which later refers queryId is find the operation log file. So, updating query Id later may have side effects.

Reply:
removed setting of query id using tag, instead added a new parameter query tag and map it to the query id. This map can be used to get the operation in case user try to kill the query using the query tag.